src/computational_graph/graph.jl

```diff
@@ -150,6 +150,20 @@ external(g::Graph) = OperatorProduct(g.vertices)[g.external]
 """
 vertices(g::Graph) = g.vertices
 
+"""
+    function external(g::Graph)
+
+    Return all physical external vertices (::Vector{OperatorProduct}) of Graph `g`.
+"""
+external(g::Graph) = OperatorProduct.(OperatorProduct(g.vertices)[g.external])
+
+"""
+    function external_with_ghost(g::Graph)
+
+    Return all the external vertices (::Vector{OperatorProduct}), including real legs and ghost.
+"""
+external_with_ghost(g::Graph) = OperatorProduct.(OperatorProduct(g.vertices)[eachindex(g.external)])
```

Comment:
I would prefer to split this function into `ghosts(g::Graph)`, and then `external_with_ghost(g::Graph) = union(ghosts(g), external(g))`.